### PR TITLE
Adsk Contrib - Python repr and LogCameraTransform fixes

### PR DESCRIPTION
**Pull request by BernardLefebvre**
Revise LogCameraTransform python bindings to address issue #1255 
Add __repr__ implementation for most python classes and create a custom __str__ implementation for Config to address issue #1256 

Signed-off-by: Bernard Lefebvre <bernard.lefebvre@autodesk.com>

### CODE REVIEW

**Review by zachlewis — APPROVED** [review]
Thank you, @BernardLefebvre !

**Comment by BernardLefebvre** [comment]
Note that the change related to __repr__ & __str__ is affecting many files, but changes there are trivial.

File: docs/api/python/frozen/pyopencolorio_gradingprimarytransform.rst [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
       Overloaded function.
 
-      1. __init__(self: PyOpenColorIO.GradingPrimaryTransform, style: PyOpenColorIO.GradingStyle = <GradingStyle.GRADING_LOG: 0>, values: PyOpenColorIO.GradingPrimary, dynamic: bool = False, dir: PyOpenColorIO.TransformDirection = <TransformDirection.TRANSFORM_DIR_FORWARD: 0>) -> None
+      1. __init__(self: PyOpenColorIO.GradingPrimaryTransform, style: PyOpenColorIO.GradingStyle = <GradingStyle.GRADING_LOG: 0>, values: OpenColorIO_v2_0rc1::GradingPrimary, dynamic: bool = False, dir: PyOpenColorIO.TransformDirection = <TransformDirection.TRANSFORM_DIR_FORWARD: 0>) -> None
```
Comment (michdolan) [inline docs/api/python/frozen/pyopencolorio_gradingprimarytransform.rst:26]: I'll do some investigation soon into why C++ types are slipping into the frozen Python docs.

**Review by michdolan — APPROVED** [review]
Thanks for all the added Python tests in here too!

File: docs/api/python/frozen/pyopencolorio_gradingprimarytransform.rst (hunk 1 above)
Comment (BernardLefebvre) [reply docs/api/python/frozen/pyopencolorio_gradingprimarytransform.rst:26]: I'll revert this and will review rst files

**Review by doug-walker — APPROVED** [review]
Received TSC approval to merge this small PR this week.

**Pull request merged**